internal/core/graphics/color.rs

```diff
@@ -192,6 +194,52 @@ impl Color {
         let rgba: RgbaColor<f32> = hsva.into();
         rgba.into()
     }
+
+    /// Returns a new version of this color with the opacity changed by _subtracting_
+    /// `amount` from it. The reference is the opacity's normalized value as `f32`.
+    ///
+    /// This means that a _positive_ `amount` will yield a more transparent color and
+    /// a _negative_ `amount` will yield a more opaque color.
+    #[must_use]
+    pub fn faded(&self, amount: f32) -> Self {
+        let mut rgba: RgbaColor<f32> = (*self).into();
+        rgba.alpha = rgba.alpha.sub(amount).clamp(0.0, 1.0);
+        rgba.into()
+    }
+
+    /// Returns a new color that is a mix of `self` and `other`, with a proportion
+    /// factor given by `factor` (which should be between `0.0` and `1.0`).
+    ///
+    /// This will make a weighted average of the corresponding RGBA channels,
+    /// with weight `factor` for `self` and `1 - factor` for `other`, and produce
+    /// the color with the resulting values.
+    #[must_use]
+    pub fn mixed(&self, other: &Self, factor: f32) -> Self {
+        // * NOTE: The implemented behavior is based on the `mix` function from SASS
```

Comment:
The first time I've seen this function was in [GLSL/OpenGL](https://registry.khronos.org/OpenGL-Refpages/gl4/html/mix.xhtml)